design-proposals/shadow-node.md

```diff
@@ -0,0 +1,328 @@
+# Overview
+
+A reported [Advisory](https://github.com/kubevirt/kubevirt/security/advisories/GHSA-cp96-jpmq-xrr2) is outlining how of our virt-handler component can be abused to escalate local privileges when a node, running virt-handler, is compromised. A flow and more details can be found on reported [issue](https://github.com/kubevirt/kubevirt/issues/9109).
+
+This proposal is outlining mitigation in a form of internal change.
+
+## Motivation
+
+Kubevirt should be secure by default. While mitigation is available, it is not part of Kubevirt.
+
+## Goals
+
+- Prevent a malicious user that has taken over a Kubernetes node where virt-handler daemonset is deployed from exploiting the daemonset's RBAC permissions in order to elevate privileges beyond the node until potentially having full privileged access to the whole cluster.
+- Prevent virt-handler from modifying non-kubevirt-owned fields on the node. This includes not being able to change the spec or any label/annotation which are not strictly kubevirt owned.
+
+## Non Goals
+
+Redesign/optimize existing virt-hanlder features to not update/patch the node in the first place.
+
+## Definition of Users
+
+Not user-facing change
+
+## User Stories
+
+Not user-facing change
+
+## Repos
+
+Kubevirt/Kubevirt
+
+# Design Options
+
+## Background
+In order to easy review and understand the different options, this section will describe different usage of Node object in the virt-handler.
+
+### Node writers on virt-handler
+Kubevirt sets ~170 labels on each node. This is set by the following entities
+1. virt-handler [labels](https://github.com/kubevirt/kubevirt/blob/5b61a18f4c3b6c549d50fd201664c21720e3dbe2/cmd/virt-handler/virt-handler.go#L185) the node with `kubevirt.io/schedulable=false` at the start and the termination (by SIGTERM only) of the virt-handler process to indicate (best-effort) that the node is not ready to handle VMs.
```

Comment:
Maybe we should mentioned that this improves user experience with scheduling.

Reply:
DONE